### PR TITLE
fix(plugins): replace ansible.utils.ipaddr with stdlib-based test plugin

### LINKED ISSUE #18: netaddr is missing in collection
## Environment

Ansible version: >= 2.15
Collection version: 1.1.3
Kubernetes distribution (k3s/kubeadm/RKE2): k3s
OS: Ubuntu

## Expected behavior
Ansible pass without error

## Actual behavior
```
ansible-playbook --inventory inventory.yml site.yml
...
TASK [cozystack.installer.cozystack : Validate master node IPs are valid addresses] **********************************************************************************************************
[DEPRECATION WARNING]: The `ansible.module_utils.common._collections_compat` module is deprecated. This feature will be removed from ansible-core version 2.24. Use `collections.abc` from the Python standard library instead.
[DEPRECATION WARNING]: Importing 'to_text' from 'ansible.module_utils._text' is deprecated. This feature will be removed from ansible-core version 2.24. Use ansible.module_utils.common.text.converters instead.
[ERROR]: Task failed: The filter plugin 'ansible.utils.ipaddr' failed: Failed to import the required Python library (netaddr) on km-cozystack-bastion-server's Python /home/ubuntu/ansible-cozystack/examples/ubuntu/kube/bin/python3. Please read the module documentation and install it in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter

Task failed.
Origin: /home/ubuntu/.ansible/collections/ansible_collections/cozystack/installer/roles/cozystack/tasks/compute-master-nodes.yml:21:3

19       Set cozystack_master_nodes explicitly in your inventory.
20
21 - name: Validate master node IPs are valid addresses
     ^ column 3

<<< caused by >>>

The filter plugin 'ansible.utils.ipaddr' failed: Failed to import the required Python library (netaddr) on km-cozystack-bastion-server's Python /home/ubuntu/ansible-cozystack/examples/ubuntu/kube/bin/python3. Please read the module documentation and install it in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter
Origin: /home/ubuntu/.ansible/collections/ansible_collections/cozystack/installer/roles/cozystack/tasks/compute-master-nodes.yml:24:9

22   ansible.builtin.assert:
23     that:
24       - (item | ansible.utils.ipaddr) != false
           ^ column 9
```

## Steps to reproduce

1. Missing
```
 # Needed for ansible.utils.ipaddr
netaddr==1.3.0
```
 

## Relevant logs

```text
```

### PR DESCRIPTION
**Pull request by lexfrei**
## Problem

The role validated master node IPs with `ansible.utils.ipaddr`, which requires the `netaddr` Python package. `ansible-galaxy collection install` does not pull `netaddr` for users, and the README never documented the requirement — so users following the docs hit a cryptic runtime failure at the validation task.

## Solution

Replace the single `ipaddr` usage with a bundled Jinja2 test plugin backed by Python's standard library `ipaddress` module. The new plugin has no external dependencies, supports both IPv4 and IPv6, and enforces the documented `type: str` contract (rejecting int/bool/bytes).

### Plugin

- Add `plugins/test/ip_address.py` — `is_ip_address` test using stdlib
- Switch `compute-master-nodes.yml` to `cozystack.installer.is_ip_address`

### Dependencies

- Drop `ansible.utils` from `galaxy.yml` and `requirements.yml`
- Drop `netaddr` pip install and `ansible.utils` install from CI

### Tests

- Unit tests at `tests/unit/plugins/test/test_ip_address.py` covering IPv4, IPv6, hostnames, malformed strings, None, int/bool/bytes rejection, and CIDR rejection
- Wire `ansible-test units` into the sanity CI job
- Add `tests/test-ipv6-inventory.yml` with documentation-range IPv6 addresses; new CI step verifies IPv6 host keys are accepted

### Versioning & docs

- Bump `galaxy.yml` to `1.2.3`, add `version_added` to plugin DOCUMENTATION
- Update `examples/*/requirements.yml` to `v1.2.3` (with v prefix matching git tags)
- Backfill CHANGELOG for v1.1.3, v1.2.1, v1.2.2 and add v1.2.3 entry
- Fix stale `cozystack_chart_version` default in README (1.1.2 → 1.2.2)

Closes #18

## Summary by CodeRabbit

* **New Features**
  * Added IPv6 host key support with dedicated test inventory fixture.
  * Introduced built-in IP address validation test.

* **Chores**
  * Removed `ansible.utils` collection dependency.
  * Updated version to v1.2.3 and default chart version to 1.2.2.

* **Tests**
  * Added unit tests for IP address validation covering IPv4, IPv6, and edge cases.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -2,6 +2,39 @@
 cozystack.installer Release Notes
 ============================
```
[the remark this answers is not in the document]
Comment (lexfrei) [reply CHANGELOG.rst]: Fixed. The title is 33 chars; both the overline and underline are now 33 `=` characters to satisfy the RST requirement.

**Review by kitsunoff — APPROVED** [review]
LGTM. Clean dependency removal — stdlib ipaddress is the right call for a single validation use case. Plugin enforces the type: str contract explicitly, unit test coverage is thorough, and CI no longer needs netaddr. Backfilled CHANGELOG entries are a nice touch.
